handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -282,8 +282,18 @@ public SslHandler(SSLEngine engine, boolean startTls, Executor delegatedTaskExec
         this.startTls = startTls;
         maxPacketBufferSize = engine.getSession().getPacketBufferSize();
 
-        wantsDirectBuffer = engine instanceof OpenSslEngine;
-        wantsLargeOutboundNetworkBuffer = !(engine instanceof OpenSslEngine);
+        boolean opensslEngine = engine instanceof OpenSslEngine;
+        wantsDirectBuffer = opensslEngine;
+        wantsLargeOutboundNetworkBuffer = !opensslEngine;
+
+        /**
+         * As JDK {@link SSLEngine#unwrap(ByteBuffer, ByteBuffer)} only works with one {@link ByteBuffer} we should not
+         * try to use a {@link CompositeByteBuf} at the first place to make it as fast as possible.
+         *
+         * When using {@link OpenSslEngine} we can call {@link OpenSslEngine#unwrap(ByteBuffer[], ByteBuffer[])} and
+         * so remove the need of memory copies for more performance.
```

Comment:
Punctuation and clarity
```
 * When using JDK {@link SSLEngine}, we use {@link #MERGE_CUMULATOR} because it works only with
 * one {@link ByteBuffer}.
 *
 * When using {@link OpenSslEngine}, we can use {@link #COMPOSITE_CUMULATOR} because it has
 * {@link OpenSslEngine#unwrap(ByteBuffer[], ByteBuffer[])} which works with multiple {@link ByteBuffer}s
 * and which does not need to do extra memory copies.
```